coloc/coloc_runner.py

```diff
@@ -0,0 +1,246 @@
+    ''',
+    )
+
+    # convert to pandas df
+    with (ro.default_converter + pandas2ri.converter).context():
+        pd_p4_df = ro.conversion.get_conversion().rpy2py(ro.r('p_df'))
+
+    # add cell type and chrom annotation to df
+    pd_p4_df['celltype'] = celltype
+    pd_p4_df['chrom'] = eqtl['chr'].iloc[0]
+
+    # write to GCS
+    pd_p4_df.to_csv(
+        coloc_results_file,
+        sep='\t',
+        index=False,
+    )
+
+
+@click.option(
+    '--egenes-files-path',
+    help='Path to the gene-level summary files',
+    default='gs://cpg-bioheart-test-analysis/saige-qtl/bioheart_n990_and_tob_n1055/241004_n100/output_files/summary_stats',
+)
+@click.option(
+    '--snp-cis-dir',
+    help='Path to the directory containing the SNP cis results',
+    default='gs://cpg-bioheart-test-analysis/saige-qtl/bioheart_n990_and_tob_n1055/241004_n100/output_files',
+)
+@click.option(
+    '--snp-gwas-file',
+    help='Path to the SNP GWAS file',
+    default='gs://cpg-bioheart-test/str/gwas_catalog/gcst/gcst-gwas-catalogs/GCST011071_parsed.tsv',
+)
+@click.option(
+    '--celltypes', help='Cell types to run, single str, comma separated', default='ASDC'
+)
+@click.option('--cis-window-size', help='Cis window size used', default=100000)
+@click.option('fdr-threshold', help='FDR threshold', default=0.05)
+@click.option(
+    '--max-parallel-jobs', help='Maximum number of parallel jobs to run', default=500
+)
+@click.option(
+    '--pheno-output-name', help='Phenotype output name', default='covid_GCST011071'
+)
+@click.option('--job-cpu', help='Number of CPUs to use for each job', default=0.25)
+@click.command()
+def main(
+    snp_cis_dir: str,
+    egenes_file_path: str,
+    celltypes: str,
+    snp_gwas_file: str,
+    pheno_output_name: str,
+    max_parallel_jobs: int,
+    cis_window_size: int,
+    fdr_threshold: float,
+    job_cpu: float,
+):
+    # Setup MAX concurrency by genes
+    _dependent_jobs: list[hb.batch.job.Job] = []
+
+    def manage_concurrency_for_job(job: hb.batch.job.Job):
+        """
+        To avoid having too many jobs running at once, we have to limit concurrency.
+        """
+        if len(_dependent_jobs) >= max_parallel_jobs:
+            job.depends_on(_dependent_jobs[-max_parallel_jobs])
+        _dependent_jobs.append(job)
+
+    # read in gene annotation file
+    var_table = pd.read_csv(
+        'gs://cpg-bioheart-test/str/240_libraries_tenk10kp1_v2/concatenated_gene_info_donor_info_var.csv',
+    )
+    hg38_map = pd.read_csv(
+        snp_gwas_file,
+        sep='\t',
+    )
+
+    b = get_batch(name=f'Run coloc:{pheno_output_name}')
+
+    for celltype in celltypes.split(','):
+
+        # read in eGenes file
+        egenes_file = (
+            egenes_file
+        ) = f'{egenes_file_path}/{celltype}_all_cis_cv_gene_level_results.tsv'
+        result_df_cfm = pd.read_csv(
+            egenes_file,
+            sep='\t',
+        )
+        result_df_cfm = result_df_cfm[
```

Comment:
so right now you're doing coloc for every egene with a eQTL that passes the FDR threshold. 

This is slightly different from me: I only colocalise for every egene with an eQTL that passes FDR AND the eTR is fine-mapped and deemed candidate causal for driving the gene.

Reply:
yes I did notice that, and was going to ask you how quick this is? I can be more stringent with the significance threshold, e.g. 5e-6 or even 5e-8 on the smallest p-value for that gene, but have not (yet?) run fine-mapping atm

Reply:
it is quick to run; takes half a day to run on everything all cell types genome-wide